database/src/test/scala/za/co/absa/atum/database/runs/GetAncestorsIntegrationTests.scala

```diff
@@ -0,0 +1,533 @@
+      .execute { queryResult =>
+        val result1 = queryResult.next()
+        assert(result1.getInt("status").get == 11)
+        assert(result1.getString("status_text").get == "Partitioning added to flows")
+      }
+
+    function(addToParentFlowsFn)
+      .setParam("i_fk_parent_partitioning", partId6)
+      .setParam("i_fk_partitioning", partId7)
+      .setParam("i_by_user", "GrandDaughter")
+      .execute { queryResult =>
+        val result1 = queryResult.next()
+        assert(result1.getInt("status").get == 11)
+        assert(result1.getString("status_text").get == "Partitioning added to flows")
+      }
+
+    function(addToParentFlowsFn)
+      .setParam("i_fk_parent_partitioning", partId3)
+      .setParam("i_fk_partitioning", partId5)
+      .setParam("i_by_user", "GrandMa")
+      .execute { queryResult =>
+        val result1 = queryResult.next()
+        assert(result1.getInt("status").get == 11)
+        assert(result1.getString("status_text").get == "Partitioning added to flows")
+      }
+
+    function(addToParentFlowsFn)
+      .setParam("i_fk_parent_partitioning", partId4)
+      .setParam("i_fk_partitioning", partId5)
+      .setParam("i_by_user", "GrandMa")
+      .execute { queryResult =>
+        val result1 = queryResult.next()
+        assert(result1.getInt("status").get == 11)
+        assert(result1.getString("status_text").get == "Partitioning added to flows")
+      }
+
+    function(addToParentFlowsFn)
+      .setParam("i_fk_parent_partitioning", partId5)
+      .setParam("i_fk_partitioning", partId8)
+      .setParam("i_by_user", "Mother")
+      .execute { queryResult =>
+        val result1 = queryResult.next()
+        assert(result1.getInt("status").get == 11)
+        assert(result1.getString("status_text").get == "Partitioning added to flows")
+      }
+
+    function(addToParentFlowsFn)
+      .setParam("i_fk_parent_partitioning", partId7)
+      .setParam("i_fk_partitioning", partId8)
+      .setParam("i_by_user", "Mother")
+      .execute { queryResult =>
+        val result1 = queryResult.next()
+        assert(result1.getInt("status").get == 11)
+        assert(result1.getString("status_text").get == "Partitioning added to flows")
+      }
+
+    //TEST 1 Ancestors Partition
+    function(getAncestorsFn)
+      .setParam("i_id_partitioning", partId3)
+      .execute { queryResult =>
+        val row = queryResult.next()
+        val returnedPartitioning = row.getJsonB("partitioning").get
+        val returnedPartitioningParsed = parse(returnedPartitioning.value)
+          .getOrElse(fail("Failed to parse returned partitioning"))
+        assert(row.getInt("status").contains(11))
+        assert(row.getString("status_text").contains("OK"))
+        assert(row.getLong("ancestorid").contains(partId1))
+        assert(returnedPartitioningParsed == expectedPartitioning1)
+        assert(row.getString("author").contains("Grandpa"))
+        assert(row.getString("author").contains("Grandpa"))
```

Comment:
also I'd recommend to test whether the result is final, by adding this on the end of this local scope:

`assert(!queryResult.next())